### PR TITLE
lightning: skip non-unique index for duplicate detection

### LINKED ISSUE #53669: lightning should not run duplicate detection on non-unique index
by the KV format of non-unique index, there's no need to check non-unique index KV overwritten because values are always empty

https://github.com/pingcap/tidb/blob/afd8de1f62188516b8bdc50c292262a20b33bd0c/pkg/lightning/backend/local/duplicate.go#L721-L725

we can skip for non-unique index here

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: close #53669 

Problem Summary:
ightning should not run duplicate detection on non-unique index.

### What changed and how does it work?
skip non-unique index for duplicate detection.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/lightning/backend/local/duplicate.go [hunk 1]
```diff
@@ -720,7 +720,7 @@ func (m *DupeDetector) buildDupTasks() ([]dupTask, error) {
 		putToTaskFunc(ranges, nil)
 	})
 	for _, indexInfo := range m.tbl.Meta().Indices {
-		if indexInfo.State != model.StatePublic {
+		if indexInfo.State != model.StatePublic || !indexInfo.Unique {
```
Comment (lance6716) [inline pkg/lightning/backend/local/duplicate.go:723]: please also check if the clustered / non-clustered PK will appear in `m.tbl.Meta().Indices` and their `.Unique`

**Review by D3Hunter — APPROVED** [review]
rest lgtm except existing comments

**Pull request closed without merging**